src/renderer/state.ts

```diff
@@ -1055,11 +1089,11 @@ export class AppState {
   /**
    * Save a key/value to localStorage.
    *
-   * @param {string} key
+   * @param {GlobalSetting | WindowSpecificSetting} key
```

Comment:
Not directly related to this PR, but I wonder how important JSDoc types are in a Typescript project (also sorry for the oversight!)

Reply:
Agreed, I'm not sure the history of how this codebase ended up with them and TypeScript, but since it does have them at the moment, I try to keep them up-to-date. 😅 I think at some point it probably makes sense to just rip out the types for params since that's duplicating with TS, but we can leave things like `@param foo - this is a description of foo` since that's still useful.